readthedocs/core/signals.py

```diff
@@ -122,10 +124,18 @@ def delete_projects(sender, instance, *args, **kwargs):
 @receiver(pre_create_historical_record)
 def add_extra_historical_fields(sender, **kwargs):
     history_instance = kwargs['history_instance']
+    if not history_instance:
+        return
+
     history_user = kwargs['history_user']
-    if history_instance and history_user:
+    if history_user:
         history_instance.extra_history_user_id = history_user.id
         history_instance.extra_history_user_username = history_user.username
 
+    request = getattr(HistoricalRecords.context, 'request', None)
```

Comment:
I think I don't understand how this works. Why `HistoricalRecords.context`, which seems to be something static, will have a dynamic `request` attribute? 🤔

Reply:
I found this example, and it seems that it's injected in the middleware. Super weird to me.

https://django-simple-history.readthedocs.io/en/latest/historical_model.html?highlight=context

Also, it seems that it should be always there and there is no need for `getattr()` and `if request`? 😕

Reply:
Yeah, it's weird how they pass the request, but that's on their docs. You can end without a request object when you save the object without hitting the middleware, this is if you save the object outside a request like a task or from the django shell.

Reply:
Good point! Thanks.